app/src/main/java/ghelani/kshamina/sssc_android_app/grades/calculator/CalculatorFragment.java

```diff
@@ -1,21 +1,164 @@
 package ghelani.kshamina.sssc_android_app.grades.calculator;
 
+import android.content.SharedPreferences;
 import android.os.Bundle;
+import android.preference.PreferenceManager;
 import android.support.annotation.NonNull;
 import android.support.annotation.Nullable;
 import android.support.v4.app.Fragment;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
+import android.widget.TextView;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.stream.Collectors;
+
+import ghelani.kshamina.sssc_android_app.DAO.CourseDao;
+import ghelani.kshamina.sssc_android_app.DAO.TermDao;
+import ghelani.kshamina.sssc_android_app.GradesDatabase;
+import ghelani.kshamina.sssc_android_app.MainActivity;
 import ghelani.kshamina.sssc_android_app.R;
+import ghelani.kshamina.sssc_android_app.entity.Course;
+import ghelani.kshamina.sssc_android_app.entity.Term;
+import ghelani.kshamina.sssc_android_app.grades.Grading;
 
 public class CalculatorFragment extends Fragment {
+
+    private TextView calculatedOverallCGPA;
+    private TextView calculatedMajorCGPA;
+
+    private List<Course> courseList = new ArrayList<>();
+    private RecyclerView recyclerView;
+    private CoursesAdapter adapter;
+    private RecyclerView.LayoutManager layoutManager;
+
+    SharedPreferences preferences;  // TODO Show/hide in-progress courses
```

Comment:
Any chance you can hook this up? The toggle is already there in the Settings tab!

Reply:
This will be done in a future PR.